azure-toolkit-libs/azure-toolkit-eventhubs-lib/src/main/java/com/microsoft/azure/toolkit/lib/eventhubs/EventHubsInstance.java

```diff
@@ -69,6 +78,55 @@ public void disableSending() {
         updateStatus(EntityStatus.SEND_DISABLED);
     }
 
+    public boolean isListening() {
+        return Objects.nonNull(this.consumerAsyncClient);
+    }
+
+    public void publishEvents(String message) {
+        final IAzureMessager messager = AzureMessager.getMessager();
+        try (final EventHubProducerClient producer = new EventHubClientBuilder()
+                .connectionString(getOrCreateConnectionString(Collections.singletonList(AccessRights.SEND)))
+                .buildProducerClient()) {
+            EventDataBatch eventDataBatch = producer.createBatch();
+            final EventData eventData =  new EventData(message);
+            if (!eventDataBatch.tryAdd(eventData)) {
+                producer.send(eventDataBatch);
+                eventDataBatch = producer.createBatch();
+                if (!eventDataBatch.tryAdd(eventData)) {
+                    throw new AzureToolkitRuntimeException("Event is too large for an empty batch. Max size: "
+                            + eventDataBatch.getMaxSizeInBytes());
+                }
+            }
+            if (eventDataBatch.getCount() > 0) {
+                producer.send(eventDataBatch);
+                messager.info(String.format("Successfully send message to event hub %s", getName()));
+            }
+        } catch (final Exception e) {
+            throw new AzureToolkitRuntimeException(e);
+        }
+    }
+
+    public void startListening() {
+        messager = AzureMessager.getMessager();
+        messager.info(String.format("Start listening to event hub %s ...", getName()));
+        this.consumerAsyncClient = new EventHubClientBuilder()
+                .connectionString(getOrCreateConnectionString(Collections.singletonList(AccessRights.LISTEN)))
+                .consumerGroup(EventHubClientBuilder.DEFAULT_CONSUMER_GROUP_NAME)
+                .buildAsyncConsumerClient();
+        this.consumerAsyncClient.receive().subscribe(partitionEvent ->
+                messager.info("Message Received: " + partitionEvent.getData().getBodyAsString()));
+    }
+
+    public void stopListening() {
+        if (Objects.isNull(this.consumerAsyncClient)) {
+            return;
+        }
+        this.consumerAsyncClient.close();
+        this.consumerAsyncClient = null;
```

Comment:
why we need to close the client instead of keeping it and re-use?

Reply:
If we re-use this client, then we should close this client in a proper time.
It seems that no proper time to close it after re-use. So, just new client when startListening and close when stopListening.